plugins/woocommerce-admin/client/products/fields/category-field/create-category-modal.tsx

```diff
@@ -0,0 +1,247 @@
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Button,
+	Modal,
+	Spinner,
+	TextControl,
+	Popover,
+} from '@wordpress/components';
+import { useMemo, useRef, useState } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import {
+	__experimentalSelectControl as SelectControl,
+	__experimentalSelectControlItem as SelectControlItem,
+} from '@woocommerce/components';
+import { recordEvent } from '@woocommerce/tracks';
+import {
+	EXPERIMENTAL_PRODUCT_CATEGORIES_STORE_NAME,
+	ProductCategory,
+} from '@woocommerce/data';
+import classnames from 'classnames';
+import { debounce } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import './create-category-modal.scss';
+import { useCategorySearch } from './use-category-search';
+import { CategoryFieldItem } from './category-field-item';
+
+type CreateCategoryModalProps = {
+	initialCategoryName?: string;
+	onCancel: () => void;
+	onCreated: ( newCategory: ProductCategory ) => void;
+};
+
+export const CreateCategoryModal: React.FC< CreateCategoryModalProps > = ( {
+	initialCategoryName,
+	onCancel,
+	onCreated,
+} ) => {
+	const {
+		categoriesSelectList,
+		isSearching,
+		categoryTreeKeyValues,
+		searchCategories,
+		getFilteredItems,
+	} = useCategorySearch();
+	const { createNotice } = useDispatch( 'core/notices' );
+	const [ isCreating, setIsCreating ] = useState( false );
+	const { createProductCategory, invalidateResolutionForStoreSelector } =
+		useDispatch( EXPERIMENTAL_PRODUCT_CATEGORIES_STORE_NAME );
+	const [ categoryName, setCategoryName ] = useState(
+		initialCategoryName || ''
+	);
+	const [ categoryParent, setCategoryParent ] =
+		useState< SelectControlItem | null >( null );
+	const selectControlMenuRef = useRef< HTMLElement >( null );
+
+	const onSave = async () => {
+		recordEvent( 'product_category_add', {
+			new_product_page: true,
+		} );
+		setIsCreating( true );
+		try {
+			const newCategory: ProductCategory = await createProductCategory( {
+				name: categoryName,
+				parent: categoryParent ? categoryParent.value : undefined,
+			} );
+			invalidateResolutionForStoreSelector( 'getProductCategories' );
+			setIsCreating( false );
+			onCreated( newCategory );
+		} catch ( e ) {
+			createNotice(
+				'error',
+				__( 'Failed to create category.', 'woocommerce' )
+			);
+			setIsCreating( false );
+			onCancel();
+		}
+	};
+
+	const onInputChange = ( searchString?: string ) => {
+		searchCategories( searchString || '' );
+	};
+
+	const searchDelayed = useMemo(
+		() => debounce( onInputChange, 150 ),
+		[ onInputChange ]
+	);
+
+	return (
+		<Modal
+			title={ __( 'Create category', 'woocommerce' ) }
+			onRequestClose={ () => onCancel() }
+			className="woocommerce-create-new-category-modal"
+		>
+			<div className="woocommerce-create-new-category-modal__wrapper">
+				<TextControl
+					label={ __( 'Name', 'woocommerce' ) }
+					name="Tops"
+					value={ categoryName }
+					onChange={ setCategoryName }
+				/>
+				<SelectControl
```

Comment:
This SelectControl should follow a similar format as the way it's used in `category-field.tsx`.
There is one main difference that we make use of `Popover` here to display the menu. This is in order to display the dropdown without part of it being hidden by the modal window.
I had to make use of `<Popover.Slot />` for this also, this is set in `category-field.tsx` under the Modal component.